src/shipchain_common/rpc.py

```diff
@@ -45,25 +45,38 @@ def call(self, method, args=None):
 
         try:
             with TimingMetric('engine_rpc.call', tags={'method': method}) as timer:
-                response_json = settings.REQUESTS_SESSION.post(self.url,
-                                                               data=json.dumps(self.payload, cls=DecimalEncoder),
-                                                               timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270)
-                                                               ).json()
+                response = settings.REQUESTS_SESSION.post(self.url, data=json.dumps(self.payload, cls=DecimalEncoder),
+                                                          timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270))
+                response_json = response.json()
```

Comment:
I think we still need to address the main issue we are having right now. If we receive a non-200 status code there is a chance that the response will be blank, or might not be json-formatted (if it's a response from nginx, for example). We should check the status code before parsing the response as json.